Dash app

APoulad opened the pull request:
This is the prototype for the Dash app, which includes a dataset downloaded as a .csv file.

APoulad committed:
initial prototype

APoulad committed:
arbitrary data for testing

APoulad committed:
Moved into folder

APoulad committed:
added date selecctor and more data

APoulad committed:
added verticle holiday lines

APoulad committed:
added labels to graph

APoulad committed:
added monthly graph and tabs

APoulad committed:
added DOY view

rameshnatarajanus commented:
Aaron, super job.  I did notice a few teeny bugs and some minor design issues which you can chase down and fix.

1. Flipping to the daily view from the monthly/year-over-year views does not seem to work.  Check your callback logic
2. Title of the graph should be "tropospheric_NO2_column_number_density (mol/m^2)".  Then remove the units  from the y-axis and just leave the magnitudes. 
3. Year-over-year view is a mess on the same plot.  Can you do a multi-plot where each facet has only one city 
4. Put a vertical line in the monthly plots at Jan so that we can see where each year begins and ends.
5. I would like to use tabs for "historical data", "model" and "forecasts" - Is there a way you can replace the tabs with radioItems as[ shown here ](https://dash.plotly.com/dash-core-components/radioitems) for the daily, monthly and year-over-plots so that all three are viewable on the same "historical data" tab?
6. Also I like the idea of annotations for the holidays on the daily graph but maybe we can avoid cluttering the screen by using some abbreviations (e.g. NYD, C, E, 4th, L, M etc.).  WDYT?

APoulad committed:
updated graph titles and labels

APoulad committed:
abbreviated holidays

APoulad committed:
added vertical lines on monthly plot

APoulad committed:
tabs changed to RadioItems

APoulad committed:
yearly view uses multiple graphs

APoulad commented:
> 1. Flipping to the daily view from the monthly/year-over-year views does not seem to work.  Check your callback logic

I could not replicate this bug as it works fine for me. Let me know if this is still and issue
> 2. Title of the graph should be "tropospheric_NO2_column_number_density (mol/m^2)".  Then remove the units  from the y-axis and just leave the magnitudes.

Changed.
> 3. Year-over-year view is a mess on the same plot.  Can you do a multi-plot where each facet has only one city

I only imagined it being used when selecting one city, but now it draws multiple plots. Take a look and let me know. I cannot move the legend around so much though.
> 4. Put a vertical line in the monthly plots at Jan so that we can see where each year begins and ends.

Added
> 5. I would like to use tabs for "historical data", "model" and "forecasts" - Is there a way you can replace the tabs with radioItems as[ shown here ](https://dash.plotly.com/dash-core-components/radioitems) for the daily, monthly and year-over-plots so that all three are viewable on the same "historical data" tab?

I have added this, but I think I could just make it a second dropdown as it looks cleaner, but let me know.
> 6. Also I like the idea of annotations for the holidays on the daily graph but maybe we can avoid cluttering the screen by using some abbreviations (e.g. NYD, C, E, 4th, L, M etc.).  WDYT?

Sounds great, I have added this.

APoulad committed:
fixed graph resizing

rameshnatarajanus reviewed (approved):
Looks good. Thanks for the super-quick work Aaron.  Look forward to getting the integration with the server side finished now,
